### ISSUE TITLE
RadialBar minAngle prop not working

### ISSUE THREAD

**Issue by bkelly-proteus** [recharts/recharts#2986]
- [ ] I have searched the [issues](https://github.com/recharts/recharts/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/simple-radial-bar-chart-forked-vmgkh8?file=/src/App.tsx)

### Steps to reproduce
See minAngle prop on RadialBar component, see the test (black) bar is less than 15 degrees

### What is expected?
Based on this https://recharts.org/en-US/api/RadialBar#minAngle I expect so see the smallest bar be 15 degrees

### What is actually happening?
The bar is significantly less than 15 degrees

| Environment | Info |
|---|---|
| Recharts | v2.0.6 |
| React | 17.0.1 |
| System | CodeSandbox |
| Browser | Chrome |

**Comment by ckifer** [comment]
This is a documentation issue. `minAngle` does not exist as a property on `RadialBar`. `minAngle` is only currently used in `Pie` - [search reference](https://github.com/recharts/recharts/search?q=minAngle). [This test](https://github.com/recharts/recharts/blob/master/test/specs/chart/RadialBarChartSpec.js#L31) is also incorrect.

**Comment by ckifer** [comment]
@akib22 this issue is addressed if the documentation site is fixed for minAngle. 

You can take on the migration of the RadialBar chart to jest and TS though if you want! That will force you to remove the incorrect prop on the test

**Comment by akib22** [comment]
Okay, I want to migrate it. Then the changes will remain in this issue or you will create a new issue for this?

**Comment by ckifer** [comment]
Created a new issue and tagged you @akib22

**Comment by keshav-gupta19** [comment]
Is the issue solved or can i work on this issue??

**Comment by ualiyou** [comment]
I looked into this on main (as of 2026-08-05) and it seems the prop is entirely absent from the implementation rather than misbehaving.

Searching the codebase, minAngle only exists for Pie:

src/polar/Pie.tsx — the redistribution logic is around L789–814
src/state/types/PieSettings.ts

There are no occurrences in src/polar/RadialBar.tsx or src/util/RadialBarUtils.tsx, and defaultRadialBarProps doesn't include it. So <RadialBar minAngle={15} /> is silently ignored, which matches the original report.

Meanwhile the docs still list it — recharts.org has a minAngle entry in src/docs/api/RadialBar.ts (default 0, marked isOptional: false). That mismatch is probably why this reads as a bug rather than a missing feature.

Two ways to close this:

Implement minAngle for RadialBar, mirroring the approach already used in Pie.
Remove it from the docs and treat it as unsupported.

I'd be happy to open a PR for either — which direction would you prefer?

**Comment by Chandu-Putta** [comment]
Hello team, thanks for posting this issue.

I cloned the repo and checked the code directly — minAngle only exists on Pie (src/polar/Pie.tsx, PieSettings.ts). RadialBar has no minAngle anywhere in its props, settings, or rendering logic. It only has minPointSize (a pixel-based minimum), which is a different thing. So the prop shown in the docs (recharts.org/en-US/api/RadialBar#minAngle) is silently ignored when passed to RadialBar — confirmed, this is a real gap, not just a docs mismatch.

I'd like to pick this up and open a PR once it's ready. Fix approach: add minAngle to RadialBar, following the same pattern already used in Pie — when a bar's angle falls below the threshold, extend its endAngle to meet the minimum, while making sure true zero-value bars are excluded from that adjustment.

There was already a good attempt at this in #7147, which never got a maintainer review and was closed from inactivity — I'll use it as a reference but build it fresh.

Will open a PR directly rather than waiting on a go-ahead here, so it's ready for review whenever someone has bandwidth.
